### PR TITLE
feat: GHA for auto updating and committing to lastest aws-sam-translator

### PR DESCRIPTION
**Pull request by jfuss**
The updating of aws-sam-translator was added to "update_app_templates_hash.yml" but renamed it so it reflected that and updating aws-sam-translator as well.

Most of the GHA was based on the "updating app templates hash" but was adapted to what this GHA needed.

#### Which issue(s) does this change fix?

none

#### Why is this change necessary?
Auto upgrade `aws-sam-translator` dependency and update testing files.

#### How does it address the issue?
Does the above

#### What side effects does this change have?
None that I know of

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: .github/workflows/automated-updates-to-sam-cli.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+name: Update aws/aws-sam-cli with latest commit hash from aws/aws-sam-cli-app-templates and aws-sam-translator version
+
+on:
+  schedule:
+    - cron: "0 0/4 * * *" # run at the top of every 4 hour
+  workflow_dispatch: {}
+
+jobs:
+  updateCommitHash:
```
Comment (moelasmar) [inline .github/workflows/automated-updates-to-sam-cli.yml:9]: can you change the job name to be more specific to the app template hash update.

File: .github/workflows/automated-updates-to-sam-cli.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      
+      - name: Checkout SAM CLI
+        uses: actions/checkout@v3
+        with:
+          repository: aws/aws-sam-cli
+          path: aws-sam-cli
+      
+      - uses: actions/setup-python@v4 # used for make update-reproducible-reqs below
+        with:
+          python-version: "3.7"
```
Comment (moelasmar) [inline .github/workflows/automated-updates-to-sam-cli.yml:76]: can we use recent python version, py3.7 will be deprecated within 2 months. Also, it is not clear where do we use python?

File: .github/workflows/automated-updates-to-sam-cli.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+          cd serverless-applicaiton-model
+          SAM_T_CUR_VERSION=$(git describe --tags --abbrev=0 | sed 's/v//')
+          cd ../aws-sam-cli
+          git checkout -b update_sam_transform_version
+          SAM_T_PRE_VERSION=$(grep "aws-sam-translator=" requirements/base.txt)
+          git reset --hard develop
+          sed -i "s/aws-sam-translator=$SAM_T_CUR_VERSION/$SAM_T_PRE_VERSION/g" requirements/base.txt; make update-reproducible-reqs
```
Comment (moelasmar) [inline .github/workflows/automated-updates-to-sam-cli.yml:88]: how can we make sure that SAM-T is already released, and it is not only an update in the SAM-T repo, and the team is still working on the release process.

File: .github/workflows/automated-updates-to-sam-cli.yml (hunk 1 above)
Comment (jfuss) [reply .github/workflows/automated-updates-to-sam-cli.yml:9]: Can update, but this was how it was originally.

File: .github/workflows/automated-updates-to-sam-cli.yml (hunk 2 above)
Comment (jfuss) [reply .github/workflows/automated-updates-to-sam-cli.yml:76]: We cannot upgrade until we update our linux installers, as they use py3.7 and the `make update-reproducible-reqs` requires 3.7 at the moment.

File: .github/workflows/automated-updates-to-sam-cli.yml (hunk 3 above)
Comment (jfuss) [reply .github/workflows/automated-updates-to-sam-cli.yml:88]: This job creates a PR. The PR tests will not pass until SAM-T is fully released to PyPi. So I think this is sufficient as is but let me know if you disagree further and we can discuss more.

**Review by moelasmar — APPROVED** [review]

**Review by hawflau — APPROVED** [review]

**Pull request merged**